endpoint.go

```diff
@@ -360,6 +368,14 @@ func (ep *endpoint) Leave(containerID string, options ...EndpointOption) error {
 		}
 	}
 
+	// Remove non-interface routes.
+	for _, r := range ep.joinInfo.StaticRoutes {
+		err = sb.RemoveStaticRoute(r)
+		if err != nil {
+			logrus.Debugf("Remove route failed: %v", err)
+		}
+	}
```

Comment:
Not sure the precise mechanism you're using to detach the interface on Leave -- but if you delete an interface, its device routes go away. Also if you delete the namespace, the interfaces and routes are all cleared too. And if you move an interface between netns, the device routes are also removed. 

So I suspect it's fine to leave them hanging -- does that sound like it will be general enough?